app/src/modules/shell/shell.c

```diff
@@ -0,0 +1,273 @@
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+#include <date_time.h>
+#include <zephyr/task_wdt/task_wdt.h>
+#include <modem/nrf_modem_lib_trace.h>
+
+#include "message_channel.h"
+
+LOG_MODULE_REGISTER(shell, CONFIG_APP_SHELL_LOG_LEVEL);
+
+/* Register subscriber */
+ZBUS_SUBSCRIBER_DEFINE(shell, CONFIG_APP_SHELL_MESSAGE_QUEUE_SIZE);
+
+enum zbus_test_type {
+	PING,
+};
+
+ZBUS_CHAN_DECLARE(ZBUS_TEST_CHAN);
+ZBUS_CHAN_DEFINE(ZBUS_TEST_CHAN,
+		 enum zbus_test_type,
+		 NULL,
+		 NULL,
+		 ZBUS_OBSERVERS(shell),
+		 ZBUS_MSG_INIT(0)
+);
+
+#define UART_DEVICE_NODE DT_CHOSEN(zephyr_shell_uart)
+static const struct device *const shell_uart_dev = DEVICE_DT_GET(UART_DEVICE_NODE);
+
+static void uart_disable_handler(struct k_work *work)
+{
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_UART
+	int err = nrf_modem_lib_trace_level_set(NRF_MODEM_LIB_TRACE_LEVEL_OFF);
+	if (err) {
+		LOG_ERR("nrf_modem_lib_trace_level_set() failed with err = %d.", err);
+	}
+#endif
+
+	if (device_is_ready(shell_uart_dev)) {
+		pm_device_action_run(shell_uart_dev, PM_DEVICE_ACTION_SUSPEND);
+	}
+}
+
+static void uart_enable_handler(struct k_work *work)
+{
+	if (device_is_ready(shell_uart_dev)) {
+		pm_device_action_run(shell_uart_dev, PM_DEVICE_ACTION_RESUME);
+	}
+
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_UART
+	int err = nrf_modem_lib_trace_level_set(NRF_MODEM_LIB_TRACE_LEVEL_FULL);
+	if (err) {
+		LOG_ERR("nrf_modem_lib_trace_level_set() failed with err = %d.", err);
+	}
+#endif
+
+	LOG_INF("UARTs enabled\n");
+}
+
+static K_WORK_DEFINE(uart_disable_work, &uart_disable_handler);
+static K_WORK_DELAYABLE_DEFINE(uart_enable_work, &uart_enable_handler);
+
+void uart_disable(void)
+{
+	LOG_INF("Disabling UARTs requested");
+	k_work_submit(&uart_disable_work);
+}
+
+void uart_enable(void)
+{
+	LOG_INF("Enabling UARTs requested");
+	k_work_schedule(&uart_enable_work, K_NO_WAIT);
+}
+
+static int cmd_uart_disable(const struct shell *sh, size_t argc,
+                         char **argv)
+{
+	int sleep_time;
+
+	if (argc != 2) {
+		LOG_ERR("disable: invalid number of arguments");
+		return -EINVAL;
+	}
+
+	sleep_time = atoi(argv[1]);
+	if (sleep_time < 0) {
+		LOG_ERR("disable: invalid sleep time");
+		return -EINVAL;
+	}
+
+	if (sleep_time > 0) {
+		LOG_INF("disable: disabling UARTs for %d seconds", sleep_time);
+	} else {
+		LOG_INF("disable: disabling UARTs indefinitely");
+	}
+	k_sleep(K_MSEC(500)); /* allow little time for printing the notification */
+	k_work_submit(&uart_disable_work);
+
+	if (sleep_time > 0) {
+		k_work_schedule(&uart_enable_work, K_SECONDS(sleep_time));
+	}
+
+	return 0;
+}
+
+static int cmd_zbus_ping(const struct shell *sh, size_t argc,
+                         char **argv)
+{
+	int err;
+	enum zbus_test_type test_type;
+	
+	ARG_UNUSED(sh);
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	test_type = PING;
+
+	err = zbus_chan_pub(&ZBUS_TEST_CHAN, &test_type, K_SECONDS(1));
+	if (err) {
+		LOG_ERR("zbus_chan_pub, error: %d", err);
```

Comment:
```suggestion
		shell_error(sh, "zbus_chan_pub, error: %d", err);
```
If you use this, `ARG_UNUSED(sh)` can be removed